### PR TITLE
Matt/160 event server action cleanup

### PR DESCRIPTION
**Pull request by Fattimo**
Changed scripts in events.ts to just return what is necessary to populate an event card on the front end.

Changed utils/types.ts to reflect the new return type.

### CODE REVIEW

File: server/actions/events.ts [hunk 1]
```diff
@@ -1,33 +1,44 @@
...
+import Nonprofit from "server/models/nonprofit";
 import errors from "utils/errors";
-import { Event as EventType } from "utils/types";
+import {
+  Event as EventType,
+  EventCardData as EventCardDataType
+} from "utils/types";
 
-export async function getUpcomingEvents() {
+const cardFields = "nonprofitId name startDate endDate name image address";
```
Comment (PC98) [inline server/actions/events.ts:10]: While this works, one problem I see is that `cardFields` and `EventCardDataType` can diverge easily. For instance, it's possible for someone to update `cardFields` but forget to make necessary changes to `EventCardDataType`. Right now you also have one extra `"name"` `cardFields` 😅. One better solution:
```
const cardFields: Record<keyof EventCardDataType, 1> = {
  nonprofitId: 1,
  name: 1,
  startDate: 1,
  endDate: 1,
  image: 1,
  address: 1
};
```

`Record` and `keyof` are also powerful TS types that I recommend looking into 🙂

File: utils/types.ts [hunk 2]
```diff
@@ -31,14 +31,17 @@ export interface Donation {
...
-export interface Event {
+export interface Event extends EventCardData {
+  about: string;
+  maxVolunteers: number;
+  volunteers: Array<string>;
+}
+
+export interface EventCardData {
   name: string;
   nonprofitId: string;
```
[the remark this answers is not in the document]
Comment (Fattimo) [reply utils/types.ts:42]: I looked up Pick, it is pretty cool! The one adjustment I made was the ability to condense it all to a single Pick like Pick<Nonprofit, "_id" | "name">

**Review by PC98 — APPROVED** [review]
💯

File: utils/types.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (PC98) [reply utils/types.ts]: That's cool, I didn't realize they were equivalent!

**Pull request merged**